### PR TITLE
Entropic Map Estimation notebook

### PR DESCRIPTION
**Pull request by APooladian**
preliminary notebook for entropic map estimation --- I can also add "debiasing" transport maps to this notebook or to a different one, whichever makes more sense

### CODE REVIEW

File: docs/notebooks/entropic_map_estimator.ipynb [hunk 1]
```diff
@@ -0,0 +1,417 @@
+{
```
Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #12.    ###closed-form transport map on the newly sampled pointed
</pre><p>Overall, it feels to me that these ### comments could be unpacked into text cells that would split code in smaller pieces</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #16.    ###compute the estimated transport map with parameter "lbd" in two steps
</pre><p>what is lbd?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #19.    out = sinkhorn(PointCloud(x, y, epsilon=varep),
</pre><p>any reason why default parameters in Sinkhorn are not used here? rather than <code>varep</code> using <code>epsilon</code>  to call the parameter would be easier to parse.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #28.    fig = plt.figure(figsize=(7,5))
</pre><p>it seems these printing routines are used more than once, maybe set a <code>def plot(....</code> and reuse it? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #30.    plt.scatter(y2[0:500,0],y2[0:500,1],c='b',marker='+',label='Target')
</pre><p>Would you call this exactly the target? it sounds a bit different than the i.i.d target sample in the first plot (but maybe I am too picky here)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <p>is T_0 an OT? if not, maybe use a different notation? if it is, then I guess some hypothesis (coordinate wise independence) should be clarified?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #9.    ###to get i.i.d samples from the "target", resample from the source, and perform the mapping
</pre><p>again this long comment would be more interesting if it were unpacked as a text cell </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #21.    plt.scatter(x[0:150,0],x[0:150,1],c='r',marker='.',label='Source')
</pre><p>same plot function should be used</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #8.    lbd = 10 ** (-1)
</pre><p>I think lbd appears here :) still, epsilon sounds simpler? and 0.1 :) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #5.    lbds = 10 ** np.arange(-2.0, 0.0,step=0.1)
</pre><p>I would go with <code>epsilons</code> :)  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #20.            ###we assume the points are sampled uniformly
</pre><p>remove this </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #34.                            a, b, threshold=0.005, lse_mode=True, jit=False,
</pre><p>again, I'd refrain from using non-default settings if possible.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='10'/>

File: docs/references.bib [hunk 2]
```diff
@@ -1,3 +1,10 @@
+@article{pooladian2021entropic,
```
Comment (michalk8) [inline docs/references.bib:1]: The citation was already included in the `references.bib` as `pooladian:21`, please remove this one use the above-mentioned.

**Review by michalk8 — CHANGES_REQUESTED** [review]

File: docs/notebooks/entropic_map_estimator.ipynb (hunk 1 above)
Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <p>Please use <code>&lt;cite data-cite="pooladian:21"&gt;&lt;/cite&gt;"</code> (also in other places) when referring to the paper, as this uses the references.bib.</p><p>I'd slightly modify the last sentence to <code>... taking advantage of the EntropicPotentials</code>class.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <p>I'd also add a section title above, e.g., <code>Setup and helper functions</code> (after extracting the plotting function as Marco suggests). </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    %pip install ott-jax
</pre><p>I'd put this on a separate cell above.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #16.    from math import *
</pre><p>I prefer no star imports</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #6.    ### data parameters
</pre><p>Minor pet-peeve: for comments, just use 1 #</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #7.    Sigma1 = jnp.eye(dim)
</pre><p>I'd prefer variable names without capitalization.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #23.    plt.scatter(x[0:150,0],x[0:150,1],c='r',marker='.',label='Source')
</pre><p>Minor pet-peeve: the <code>0</code> in <code>0:150</code> is not really necessary (also in other places).  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #22.    entpots = EntropicPotentials(out.f, out.g, out.geom,a,b)
</pre><p>You can use <code>out.to_dual_potentials</code> to create the <code>EntropicPotentials</code> object.  </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #8.    plt.loglog(x,y_means,label=r'$\hat{T}_\epsilon$',c=c1,linewidth=1.5)
</pre><p>Label is added, but no legend shown, think <code>plt.legend()</code> should fix this. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #12.    plt.title(r'MSE as a function of $\varepsilon$ for $n=${} with $d=${}' .format(n,dim))
</pre><p>Please use f-strings instead of <code>format</code>.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <p>Remove empty code cell.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (marcocuturi) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <p>I think it would also help to have a formal definition of $f_\varepsilon$ somewhere, linked with the output of Sinkhorn.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (michalk8) [inline docs/notebooks/entropic_map_estimator.ipynb:1]: <p>The <code>x \in R^d</code> part doesn't render properly in the notebook. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/ott-jax/ott/pull/175/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

**Comment by michalk8** [comment]
Hey @APooladian , what's the status of this PR? Most of the requested changes are cosmetic and can finish them quickly if you prefer.
